plugins/kubernetes/test/models/kubernetes/template_filler_test.rb

```diff
@@ -556,6 +555,24 @@ def add_init_container(container)
         template.to_hash.dig_fetch(:spec, :scaleTargetRef, :name).must_equal("test-app-server")
       end
     end
+
+    describe "blue-green" do
+      before { doc.kubernetes_release.deploy.stage.blue_green = true }
+
+      it "modifies the service" do
+        raw_template[:kind] = 'Service'
+        template.to_hash.dig_fetch(:spec, :selector, :blue_green).must_equal 'green'
```

Comment:
The very first deploy should be 'blue'. Maybe it's green now because we're not flipping `blue_phase` in https://github.com/zendesk/samson/pull/2473/commits/0bfb637a67b9559c1a9867376d10ae83a68f0a58#diff-aa5b7e4c40a65c9a9a4b56d6b66e9d5aR22